QWeb/internal/text.py

```diff
@@ -104,13 +105,23 @@ def get_text_elements(text: str, **kwargs) -> Optional[list[WebElement]]:
             web_elements = _get_contains_text_element(text, **kwargs)
         except NoSuchFrameException:
             logger.debug("Got no such frame from contains text")
+    # find text directly from slots with supported parent elements
+    xpath_elements = None
+    slot_elements = get_slot_elements(text, partial)
+    if slot_elements:
+        xpath_elements = element.get_visible_elements_from_elements(slot_elements, **kwargs) or None
+    if xpath_elements:
+        if web_elements is None:
```

Comment:
I believe this can cause `UnboundLocalError` if both L99 and L105 raise an error.
On L96 where  `web_elements` is defined it should be initialized with `None`

Reply:
not sure since called functions return None if nothing is found. Maybe if some exception occurs but those should be catched anyway. But I'll add it "just in case"

Reply:
handled with new commit